### PR TITLE
Generate README with GH Actions

### PR DESCRIPTION
**Pull request by thomasmfish**

### CODE REVIEW

File: docs/template.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+- Easy to use
+- Open source
+- Reproducable results
+- Can be used as part of an automatic processing pipeline
+
+### Current state
+
+Under development
+
+{{ SECTIONCONFIGURATION_MD }}
```
Comment (PeterC-DLS) [inline docs/template.md]: Typo here?

File: docs/template.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+This is built for easy use from the command line or as part of an auto-processing pipeline, allowing parameters to be pre-configured or defined at runtime.
+
+### Goal
+
+To reconstruct the SIM data and create OTFs from PSFs from the cryoSIM at B24, Diamond from the .dv files created by Cockpit. Requirements are:
+
+- Easy to use
+- Open source
+- Reproducable results
```
Comment (PeterC-DLS) [inline docs/template.md]: Reproducible

File: .github/workflows/generate_README.yaml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+      contents: write
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          ref: ${{ inputs.BRANCH_NAME }}
+      - name: Fill template
+        working-directory: "docs/"
+        timeout-minutes: 0.1
+        run: |
+          sub_doc_parts_from() {
```
Comment (PeterC-DLS) [inline .github/workflows/generate_README.yaml]: I would put this in a script in a releng directory as it's easier to test

File: .github/workflows/generate_help_docs.yaml [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+          if [ ! -f "README.md" ]; then
+            touch README.md
+          fi
+      - name: Get help strings from package
+        run: |
+          mkdir -p docs/help
+          python -m sim_recon.cli.parsing.otf '--help' >docs/help/sim_otf.txt
+          python -m sim_recon.cli.parsing.recon '--help' >docs/help/sim_recon.txt
+          python -m sim_recon.cli.parsing.otf_view '--help' >docs/help/otf_view.txt
+      - name: Setup git
```
Comment (PeterC-DLS) [inline .github/workflows/generate_help_docs.yaml]: Perhaps use https://github.com/stefanzweifel/git-auto-commit-action

Comment (thomasmfish) [reply .github/workflows/generate_help_docs.yaml]: Okay, seems to do a good job, and I was able to remove the branch name bits too!

File: .github/workflows/generate_README.yaml (hunk 3 above)
Comment (thomasmfish) [reply .github/workflows/generate_README.yaml]: Done

**Comment by thomasmfish** [comment]
Going to stop fiddling there - this seems to work well now will fix some of the issues with the README in #17

File: releng/README_from_docs.sh [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+            echo "Saving $file to $placeholder with expanded environment variables" >&2
+            # Export file contents to text following a pattern in template.md, e.g.:
+            # section/test_file.md would replace "{{ SECTION_TEST_FILE_MD }}""
+            template="${template//"$placeholder"/"$(<"$file")"}"
+        fi
+    done
+    echo "$template"
+}
+
+REPO_DIR="$(cd -- "$(dirname -- "${BASH_SOURCE[0]}")/.." &>/dev/null && pwd)"
```
Comment (PeterC-DLS) [inline releng/README_from_docs.sh]: I think [realpath](https://manpages.ubuntu.com/manpages/jammy/en/man1/realpath.1.html) is available in the Ubuntu runner

**Review by PeterC-DLS — APPROVED** [review]

**Pull request merged**